src/docfx/build/legacy/LegacyXrefMap.cs

```diff
@@ -11,7 +11,10 @@ internal static class LegacyXrefMap
         public static void Convert(Docset docset, Context context, XrefMap xrefMap)
         {
             var map = new XrefMapModel();
-            map.References.AddRange(xrefMap.References);
+            if (xrefMap.References != null)
```

Comment:
instead of fixing it here, can you make `XrefMap.References` always non-nullable? all of our collection types should be non-nullable to avoid checking arrays against null everywhere. 

Use an empty dictionary [here](https://github.com/dotnet/docfx/blob/v3/src/docfx/build/xref/XrefMap.cs#L47) if `BuildInternalXrefMap` is false

Also `XrefMap.References` -> `XrefMap.InternalReferences` to be more explicit.

Reply:
sounds better, will do